src/python/WMCore/Database/ExecuteDAO.py

```diff
@@ -0,0 +1,287 @@
+            config.CoreDatabase.socket = os.getenv("DBSOCK", None)
+
+        # check if all database connection parameters are provided:
+        if not all([getattr(config.CoreDatabase, "connectUrl", None),
+                    getattr(config.CoreDatabase, "dialect", None),
+                    getattr(config.CoreDatabase, "socket", None)]):
+            raise RuntimeError("You must set proper DATABASE parameters: connectUrl, dialect, socket!")
+
+        # Connecting to database:
+        self.init.setDatabaseConnection(config.CoreDatabase.connectUrl,
+                                        config.CoreDatabase.dialect,
+                                        config.CoreDatabase.socket)
+
+        self.dbi = myThread.dbi
+        self.package = package
+        self.daoModule = daoModule
+
+        # Avoid any name that starts with _
+        self.sqlRegEx = re.compile("^(?!_.*)", re.IGNORECASE)
+
+        self.daoFactory = DAOFactory(package=package,
+                                     logger=self.logger,
+                                     dbinterface=self.dbi)
+        self.logger.info("DAO Factory initialised from package: %s", self.package)
+
+        self.dao = self.daoFactory(classname=daoModule)
+        self.logger.info("DAO Module initialised as: %s", self.daoModule)
+
+    def __call__(self, *sqlArgs, dryRun=False, daoHelp=False, **sqlKwArgs):
+        """
+        __call__
+        The ExecuteDAO call method. This is the method to forward all provided
+        arguments to the execute method of the DAO and return the result from the query
+        :param dryRun:      Bool flag to indicate dryrun method
+        :param *sqlArgs:    All positional arguments to be forwarded to the DAO's execute method.
+        :param **sqlKwArgs: All named arguments to be forwarded to the DAO's execute method.
+        :return:            The result from the DAO. Depending on the DAO itself it Could be one of:
+                            * A dictionary
+                            * A list
+                            * A generator
+        """
+        if dryRun:
+            results = []
+            if daoHelp:
+                self.getHelp()
+            self.logger.info("DAO SQL queries to be executed:")
+            sqlQueries = self.getSqlQuery()
+            for sqlName, sqlStr in sqlQueries.items():
+                msg = "\n----------------------------------------------------------------------\n"
+                msg += "%s: %s"
+                msg += "\n----------------------------------------------------------------------\n"
+                self.logger.info(msg, sqlName, sqlStr)
+            self.logger.info("DAO SQL arguments provided:\n%s, %s", pformat(sqlArgs), pformat(sqlKwArgs))
+        else:
+            results = self.dao.execute(*sqlArgs, **sqlKwArgs)
+            self.logger.info("DAO Results:\n%s", pformat(results if isinstance(results, dict) else list(results)))
```

Comment:
Why are you casting results to a list here? Is it a generator?

Reply:
The result could be one of the three:
 * a dictionary - in which case it is printed as it is
 * a generator - in which case it is casted to a list
 * a list - in which case the cast does nothing